packages/template-retail-react-app/app/components/product-tile/index.jsx

```diff
@@ -332,7 +362,11 @@ ProductTile.propTypes = {
      * as a swatch below the main image. The default for this property is `color`.
      */
     selectableAttributeId: PropTypes.string,
-    dynamicImageProps: PropTypes.object
+    dynamicImageProps: PropTypes.object,
+    /**
+     * Badge labels to display on the tile.
+     */
+    badgeLabels: PropTypes.array
```

Comment:
Sorry this comment got lost somewhere in my plethora of tabs. 

I feel like "badgeLabels" is a misnomer. Given it's name and it's jsDoc comment, I would assume that all the items in the array being passed would show up on the component. But this isn't the case. The prop here is actually a type of allow list that filters the custom types that are on the product object.

We could do 2 things:

1. Make the component dumb and `badgeLabels` does what it is described currently which is all the items in the list  are shown. We would have to move the logic to the product list component. (Although I think that this is the best practice for react component dev, I don't think it fits our current pattern for this component)
2. Remove the prop 'badgesLabels' and let the component be smart about what it does. This is our current pattern as we are passing in the product as a prob. Configuraton of what is shown will be done as we currently do by updating the constant. (We should also rename the constant so it's more self explainatory)
3. We rename the prop to something that is more easily understood and update the jsDoc to align with this prop. 

If you wanna talk about this, reach out to me and we can chat.

Reply:
I think no. 3 might be the way to go here.

Reply:
I think we can go with option #2. I also see a pattern where we send some of constants to product tile as props.